sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1995,8 +2020,7 @@ pi_result piQueueRetain(pi_queue Queue) {
 pi_result piQueueRelease(pi_queue Queue) {
   PI_ASSERT(Queue, PI_INVALID_QUEUE);
 
-  // Lock automatically releases when this goes out of scope.
-  std::lock_guard<std::mutex> lock(Queue->PiQueueMutex);
+  Queue->PiQueueMutex.lock();
```

Comment:
please remind why you can't keep scoped lock here? (i think it will remember the mutex in constructor, and not access it in the deleted queue again in destructor).

Reply:
std::lock_guard is created using Queue->PiQueueMutex, and this mutex is still required when it is releasing the lock_guard at the end of the scope (end of the function).
But, we already added "delete Queue" which already removed PiQueueMutex data member and it causes segmentation fault. So, the only option for me is to lock() and unlock() on the mutex itself.

Reply:
I see. Can we maybe do smth like this:

{
need_delete = false
lock_gaurd()
  if (...) {
    need_delete = true
  }
}
if (need_delete) delete;

Reply:
That's a good idea. 👍 
So, we can limit the lock_guard only to decrease and check the RefCount.
And then immediately release before we remove the Queue.
That's awesome. Thanks.

Reply:
Done. Thanks.